### PR TITLE
add stacked bilm

### PR DESCRIPTION
**Pull request by DeNeutoy**
Adds a bidirectional LSTM with variational dropout.

@matt-gardner you should try this in the parser - it will probably get you another point.

### CODE REVIEW

File: allennlp/modules/stacked_bidirectional_lstm.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
+"""
+A stacked LSTM with LSTM layers which alternate between going forwards over
```
Comment (matt-gardner) [inline allennlp/modules/stacked_bidirectional_lstm.py]: You can probably just remove this comment - it's for a different file, anyway.

**Comment by DeNeutoy** [comment]
@matt-gardner this is ready.

File: allennlp/modules/stacked_bidirectional_lstm.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import torch
+from torch.nn.utils.rnn import PackedSequence, pack_padded_sequence, pad_packed_sequence
+from allennlp.modules.augmented_lstm import AugmentedLstm
+from allennlp.common.checks import ConfigurationError
+
+
+class StackedBidirectionalLstm(torch.nn.Module):
+    """
+    A standard stacked Bidirectional LSTM where the LSTM layers
+    are concatenated between each layer.
```
Comment (matt-gardner) [inline allennlp/modules/stacked_bidirectional_lstm.py]: You should probably note what the differences are between this and the standard LSTM, and why you might want to use one or the other.

File: allennlp/modules/stacked_bidirectional_lstm.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    num_layers : int, required
+        The number of stacked Bidirectional LSTMs to use.
+    recurrent_dropout_probability: float, optional (default = 0.0)
+        The dropout probability to be used in a dropout scheme as stated in
+        `A Theoretically Grounded Application of Dropout in Recurrent Neural Networks
+        <https://arxiv.org/abs/1512.05287>`_ .
+
+    Returns
+    -------
+    output_accumulator : PackedSequence
```
Comment (matt-gardner) [inline allennlp/modules/stacked_bidirectional_lstm.py]: This doesn't match what you have in `forward`.  Do you even need this here at all?

**Comment by Mrs-Hudson** [comment]
I was wondering why this module does not use CUDA?
